server/src/weaverbird/backends/pypika_translator/translators/base.py

```diff
@@ -562,16 +589,43 @@ def _get_single_condition_criterion(
                         case RegexOp.REGEXP:
                             return column_field.regexp(condition.value).negate()
                         case RegexOp.SIMILAR_TO:
+                            compliant_regex = (
+                                compliant_regex.replace('%', '%%')
+                                if self.DIALECT in [SQLDialect.POSTGRES, SQLDialect.REDSHIFT]
+                                else compliant_regex
+                            )
```

Comment:
If this is intended for escaping, it does not work, an `ESCAPE` statement must be used:

```
playground_db=# select "Transaction_date", "City" from sales WHERE "sales"."State" SIMILAR TO '%Aviv';
  Transaction_date   |   City
---------------------+----------
 2009-01-04 13:17:00 | Tel Aviv
(1 row)

playground_db=# select "Transaction_date", "City" from sales WHERE "sales"."State" SIMILAR TO '%%Aviv';
  Transaction_date   |   City
---------------------+----------
 2009-01-04 13:17:00 | Tel Aviv
(1 row)

playground_db=# select "Transaction_date", "City" from sales WHERE "sales"."State" SIMILAR TO '%Aviv' ESCAPE '%';
 Transaction_date | City
------------------+------
(0 rows)
```

Reply:
Thanks! I still don't understand why double `%` are required for postgres and redshift for `SIMILAR TO` though :thinking: